lisk-dex-electron/src/App.jsx

```diff
@@ -525,22 +527,26 @@ class App extends React.Component {
       minAsk,
       yourOrders: Object.values(yourOrderMap),
     };
-
-    this.setState(newState);
+    return newState;
   }
 
   getPriceDecimalPrecision() {
     const { dexOptions } = this.state.configuration.markets[this.state.activeMarket];
     return dexOptions.priceDecimalPrecision == null ? DEFAULT_PRICE_DECIMAL_PRECISION : dexOptions.priceDecimalPrecision;
   }
 
+  _updateUIWithNewData = () => {
+    Promise.all([this.refreshOrderbook(), this.refreshPriceHistory()]).then(([newOrderBookState, newPriceHistoryState]) => {
```

Comment:
We should avoid using `then` for consistency.
Instead we can make the `_updateUIWithNewData` function `async` and then do:

```
const [newOrderBookState, newPriceHistoryState] = await Promise.all([this.refreshOrderbook(), this.refreshPriceHistory()])
```